climakitae/threshold_tools.py

```diff
@@ -0,0 +1,613 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+from scipy import stats
+
+import lmoments3 as lm
+from lmoments3 import distr as ldistr
+from lmoments3 import stats as lstats
+
+import matplotlib.pyplot as plt
+from matplotlib import cm
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import geoviews as gv
+
+import holoviews as hv
+from holoviews import opts
+import hvplot.pandas
+import hvplot.xarray
+
+from .visualize import get_frequency_plot, get_geospatial_plot
+
+#####################################################################
+
+
+def get_ams(da, extremes_type="max"):
+
+    extremes_types = ["max"]
+    if extremes_type not in extremes_types:
+        raise ValueError(
+            "invalid extremes type. expected one of the following: %s" % extremes_types
+        )
+
+    if extremes_type == "max":
+        ams = da.resample(time="A").max(keep_attrs=True)
+        ams.attrs["extreme value extraction method"] = "block maxima"
+        ams.attrs["extremes type"] = "maxima"
+        ams.attrs["block size"] = "1 year"
+        ams.attrs["timeseries type"] = "annual max series"
+
```

Comment:
For training @gdicecco documentation should note limitations of annual block maxima approach, i.e. that a lot of extreme events are excluded, or said differently you could have 5 extremes in one year each of which is more extreme than the peak event in subsequent years (important for california atmospheric river events, wet years vs dry years)

Reply:
propose adding Peaks over Threshold approach to backlog